src/librustc_codegen_llvm/lib.rs

```diff
@@ -120,11 +120,11 @@ impl ExtraBackendMethods for LlvmCodegenBackend {
     fn write_metadata<'b, 'gcx>(
         &self,
         tcx: TyCtxt<'b, 'gcx, 'gcx>,
-        metadata: &ModuleLlvm
+        metadata: &mut ModuleLlvm
     ) -> EncodedMetadata {
         base::write_metadata(tcx, metadata)
```

Comment:
`write_metadata` should also take `&mut ModuleLlvm` to properly signal the mutability.